shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -378,6 +378,68 @@ - (BOOL)isEqualTo:(FlutterTextRange*)other {
 }
 @end
 
+#pragma mark - FlutterTokenizer
+
+@implementation FlutterTokenizer {
+  FlutterTextInputView* _textInputView;
+}
+
+- (instancetype)initWithTextInput:(UIResponder<UITextInput>*)textInput {
+  NSAssert([textInput isKindOfClass:[FlutterTextInputView class]],
+           @"The FlutterTokenizer can only be used in a FlutterTextInputView");
+  self = [super initWithTextInput:textInput];
+  if (self) {
+    _textInputView = (FlutterTextInputView*)textInput;
+  }
+  return self;
+}
+
+- (UITextRange*)rangeEnclosingPosition:(UITextPosition*)position
+                       withGranularity:(UITextGranularity)granularity
+                           inDirection:(UITextDirection)direction {
+  UITextRange* result;
+  switch (granularity) {
+    case UITextGranularityLine:
+      // The default UITextInputStringTokenizer does not handle line granularity
+      // correctly. We need to implement our own line tokenizer.
+      result = [self lineEnclosingPosition:position];
+      break;
+    case UITextGranularityCharacter:
+    case UITextGranularityWord:
+    case UITextGranularitySentence:
+    case UITextGranularityParagraph:
+    case UITextGranularityDocument:
```

Comment:
Intentional fallthroughs should be documented with a comment: https://google.github.io/styleguide/objcguide.html#conditionals